### PR TITLE
[YUNIKORN-2414] Embeds fontawesome and roboto

### PR DESCRIPTION
**Pull request by dcoric**
### What is this PR for?

This PR embeds Fontawesome and Google Roboto as part of the project. 
This allows web UI to run in internal networks without access to the public.

### What type of PR is it?

[x] - Improvement

### What is the Jira issue?
[YUNIKORN-2414](https://issues.apache.org/jira/browse/YUNIKORN-2414)

### CODE REVIEW

**Comment by wilfred-s** [comment]
We need to check the license of the inclusions. Not all licenses are allowed: https://www.apache.org/legal/resolved.html

the pacakge typeface-roboto has been deprecated and we should move to fontsource-roboto as per the readme: https://classic.yarnpkg.com/en/package/typeface-roboto#readme

- material-design-icons is Apache 2.0 (**OK**)
- fontsource-roboto is Apache 2.0 (**OK**)
- fontawesome is mixed MIT / CC / SIL OFL 1.1 License (class B,  **maybe OK** in binary form)

What are we using from the fontawesome-free package? Icons or fonts

**Comment by dcoric** [comment]
This is a tricky one. We are using icons. The way we are using them is with CSS classes to render them from the font file. I had to look this up and this is what I got about the subject:

> When you use CSS classes to display icons from a library like Font Awesome, the license that applies would be the one covering the use of the icons themselves.
In the case of Font Awesome Free, the icons are licensed under the Creative Commons Attribution 4.0 International license (CC BY 4.0). So, even if you are using CSS classes to display those icons, the CC BY 4.0 license still applies.
This means you are free to use, share, and adapt those icons for any purpose, including commercially, as long as you provide appropriate attribution.

**Comment by wilfred-s** [comment]
Based on the changes we are only including the icons in the "convenience binary", i.e. the docker images, that we build.
We do need attribution of what we include, which means a NOTICE file etc in our docker images that we build. This would require a change in the way we build the image. It does not stop us from doing this.

In short: yes we can do this, but not 2 days out of the release branching etc for YuniKorn 1.5.
Tagging the jira for YuniKorn 1.6, we can commit the change after we branch.

File: package.json [hunk 1]
```diff
@@ -34,17 +34,20 @@
...
+    "angular-material-expansion-panel": "^0.7.2",
     "chart.js": "^4.4.0",
     "chartjs-adapter-date-fns": "^3.0.0",
     "date-fns": "^2.30.0",
+    "material-design-icons": "^3.0.1",
     "moment": "^2.29.4",
     "ngx-spinner": "^16.0.2",
     "rxjs": "~7.8.1",
     "tslib": "^2.6.2",
+    "typeface-roboto": "^1.1.13",
```
Comment (wilfred-s) [inline package.json:50]: the pacakge typeface-roboto has been deprecated and we should move to fontsource-roboto as per the [readme](https://classic.yarnpkg.com/en/package/typeface-roboto#readme)

**Review by wilfred-s — CHANGES_REQUESTED** [review]

**Comment by craigcondit** [comment]
> Based on the changes we are only including the icons in the "convenience binary", i.e. the docker images, that we build. We do need attribution of what we include, which means a NOTICE file etc in our docker images that we build. This would require a change in the way we build the image. It does not stop us from doing this.

I don't think this requires any changes at all. One of the features of Angular is an auto-generated 3rdpartylibraries.txt file at the root of the generated content -- I've checked using this PR, and fontawesome gets included there along with the appropriate license text, so I believe we are covered on the attribution front. You can even pull this up from a running yunikorn-web instance via http://localhost:9889/3rdpartylicenses.txt

File: package.json (hunk 1 above)
Comment (dcoric) [reply package.json:50]: Thanks, I will update the PR

**Review by craigcondit — APPROVED** [review]
Looks good; will approve assuming pre-commit checks pass now. I've manually updated puppeteer to 22.2.0 to hopefully fix the chromium download issue.

**Pull request closed without merging**